### PR TITLE
Updated type guessers to use FQCNs for guessed types

### PR DESCRIPTION
**Pull request by hiddewie**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #16332 |
| License | MIT |
| Doc PR | symfony/symfony-docs#5834 (if applicable) |

### CODE REVIEW

**Comment by Tobion** [comment]
You forgot https://github.com/hiddewie/symfony/blob/typeGuesser/src/Symfony/Bridge/Doctrine/Form/DoctrineOrmTypeGuesser.php#L51 and please also update the test in https://github.com/symfony/symfony/blob/2.8/src/Symfony/Component/Form/Tests/FormFactoryTest.php#L588 and https://github.com/symfony/symfony/blob/2.8/src/Symfony/Component/Form/Tests/FormFactoryTest.php#L597

**Comment by hiddewie** [comment]
Excuse me, I had overlooked those lines.

File: src/Symfony/Component/Form/Extension/Validator/ValidatorTypeGuesser.php [hunk 1]
```diff
@@ -91,79 +91,79 @@ public function guessTypeForConstraint(Constraint $constraint)
...
+                return new TypeGuess('Symfony\Component\Form\Extension\Core\Type\EmailType', array(), Guess::HIGH_CONFIDENCE);
 
             case 'Symfony\Component\Validator\Constraints\File':
             case 'Symfony\Component\Validator\Constraints\Image':
-                return new TypeGuess('file', array(), Guess::HIGH_CONFIDENCE);
+                return new TypeGuess('Symfony\Component\Form\Extension\Core\Type\FileType', array(), Guess::HIGH_CONFIDENCE);
 
             case 'Symfony\Component\Validator\Constraints\Language':
-                return new TypeGuess('language', array(), Guess::HIGH_CONFIDENCE);
+                return new TypeGuess('Symfony\Component\Form\Extension\Core\Type\LamguageType', array(), Guess::HIGH_CONFIDENCE);
```
Comment (Tobion) [inline src/Symfony/Component/Form/Extension/Validator/ValidatorTypeGuesser.php]: typo

File: src/Symfony/Component/Form/Extension/Validator/ValidatorTypeGuesser.php [hunk 2]
```diff
@@ -91,79 +91,79 @@ public function guessTypeForConstraint(Constraint $constraint)
...
             case 'Symfony\Component\Validator\Constraints\Count':
-                return new TypeGuess('collection', array(), Guess::LOW_CONFIDENCE);
+                return new TypeGuess('Symfony\Component\Form\Extension\Core\Type\CollectionType', array(), Guess::LOW_CONFIDENCE);
 
             case 'Symfony\Component\Validator\Constraints\True':
             case 'Symfony\Component\Validator\Constraints\False':
             case 'Symfony\Component\Validator\Constraints\IsTrue':
             case 'Symfony\Component\Validator\Constraints\IsFalse':
-                return new TypeGuess('checkbox', array(), Guess::MEDIUM_CONFIDENCE);
+                return new TypeGuess('Symfony\Component\Form\Extension\Core\Type\CheckBoxType', array(), Guess::MEDIUM_CONFIDENCE);
```
Comment (Tobion) [inline src/Symfony/Component/Form/Extension/Validator/ValidatorTypeGuesser.php]: CheckboxType

Are you writing these classes by hand...?

File: src/Symfony/Bridge/Doctrine/Form/DoctrineOrmTypeGuesser.php [hunk 3]
```diff
@@ -48,35 +48,35 @@ public function guessType($class, $property)
             $multiple = $metadata->isCollectionValuedAssociation($property);
             $mapping = $metadata->getAssociationMapping($property);
 
-            return new TypeGuess('entity', array('em' => $name, 'class' => $mapping['targetEntity'], 'multiple' => $multiple), Guess::HIGH_CONFIDENCE);
+            return new TypeGuess('Symfony\Component\Form\Extension\Core\Type\EntityType', array('em' => $name, 'class' => $mapping['targetEntity'], 'multiple' => $multiple), Guess::HIGH_CONFIDENCE);
```
Comment (Tobion) [inline src/Symfony/Bridge/Doctrine/Form/DoctrineOrmTypeGuesser.php]: Symfony\Bridge\Doctrine\Form\Type\EntityType

**Comment by Tobion** [comment]
:+1: 

Status: Reviewed

**Comment by jakzal** [comment]
:+1:

**Comment by Tobion** [comment]
Thank you @hiddewie.

**Pull request closed without merging**